lib/permutations.rb

```diff
@@ -1,4 +1,27 @@
+# Time complexity: O(n) => Because it is not a nested loop and it loops through depending on the input size. It changes from O(2n) to O(n) after dropping a constant, 2.
+
+# Space complexity: O(n) => Because one hash was created. 
 
 def permutations?(string1, string2)
```

Comment:
Correct on time/space complexity, but not quite a perfect answer.   Consider `heelo` and `hello`.

Reply:
Thanks Chris!  👍 I modified my code based on your comments. I hope I got it right this time. 😄